### PR TITLE
python3Packages.starlette: fix build

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Fix build failure caused by failing test (https://github.com/encode/starlette/issues/1131).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 111079` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>python39Packages.starlette</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python37Packages.fastapi</li>
    <li>python37Packages.starlette</li>
    <li>python38Packages.fastapi</li>
    <li>python38Packages.starlette</li>
  </ul>
</details>

**Comment by fabaff** [comment]
This will need another round as Python 3.9 is failing.

File: pkgs/development/python-modules/starlette/default.nix
Comment (mweinelt) [inline pkgs/development/python-modules/starlette/default.nix:50]: Coverage is another thing we don't really need to test in nixpkgs.

**Comment by mweinelt** [comment]
I'm already waiting for the uvloop release for 3.9 for a while, I don't think we should wait with this change until then.

File: pkgs/development/python-modules/starlette/default.nix
Comment (mweinelt) [reply pkgs/development/python-modules/starlette/default.nix:50]: Patch out these lines in postPatch instead: https://github.com/encode/starlette/blob/ea1990415ebef516e17f2cb724a1c4384c479597/setup.cfg#L20-L22

File: pkgs/development/python-modules/starlette/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/starlette/default.nix:55]:
```suggestion
  disabledTestFiles = [ "tests/test_graphql.py" ];
```

File: pkgs/development/python-modules/starlette/default.nix
Comment (dotlambda) [reply pkgs/development/python-modules/starlette/default.nix:50]: What's so bad about having `pytest-cov` in `checkInputs` if it saves us from patching?

File: pkgs/development/python-modules/starlette/default.nix
Comment (mweinelt) [reply pkgs/development/python-modules/starlette/default.nix:55]: Do note, that until #112095 is merged this will not work for multiple files or file paths that have spaces.

File: pkgs/development/python-modules/starlette/default.nix
Comment (mweinelt) [reply pkgs/development/python-modules/starlette/default.nix:50]: It's just not helpful and adds noise to the build process and the patching would be quite stable in most cases.

```
sed -i "s/--cov/d" setup.cfg
```

**Comment by mweinelt** [comment]
@fabaff Can you please update this pull request?

File: pkgs/development/python-modules/starlette/default.nix
Comment (fabaff) [reply pkgs/development/python-modules/starlette/default.nix:50]: Added, thanks.

File: pkgs/development/python-modules/starlette/default.nix
Comment (fabaff) [reply pkgs/development/python-modules/starlette/default.nix:55]: Switched to `disabledTestFiles`.

File: pkgs/development/python-modules/starlette/default.nix
Comment (fabaff) [reply pkgs/development/python-modules/starlette/default.nix]: In fact that will be needed for 0.14.2 but not for 0.13.8.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 111079` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>python39Packages.starlette</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python37Packages.fastapi</li>
    <li>python37Packages.starlette</li>
    <li>python38Packages.fastapi</li>
    <li>python38Packages.starlette</li>
  </ul>
</details>

File: pkgs/development/python-modules/starlette/default.nix
Comment (mweinelt) [inline pkgs/development/python-modules/starlette/default.nix:54]: This issue was closed 10 days ago. Is the test working now?

Comment (mweinelt) [reply pkgs/development/python-modules/starlette/default.nix]: I updated the package to 0.14.2 and dropped the disable.

**Comment by mweinelt** [comment]
@GrahamcOfBorg build python3Packages.fastapi

**Comment by mweinelt** [comment]
@GrahamcOfBorg build python3Packages.fastapi

File: pkgs/development/python-modules/starlette/default.nix
Comment (mweinelt) [reply pkgs/development/python-modules/starlette/default.nix:54]: Nevermind, this breaks fastapi because they *still* haven't gotten rid of UJSONResponse.

**Review by mweinelt — APPROVED** [review]

**Pull request merged**
